Add support for AWS Polly service

Gringill opened the pull request:
https://aws.amazon.com/polly/

Added support for all Polly operations
http://docs.aws.amazon.com/polly/latest/dg/API_Operations.html

Added integration and operation-struct-building tests.

Let me know what else I need to do to make this happen :).

benwilson512 commented on lib/ex_aws/operation/polly.ex:
```diff
@@ -0,0 +1,57 @@
+defmodule ExAws.Operation.Polly do
```
Operations are not per service, but rather per category of service. Polly is a `RestJSON` service. There is not at the moment a RestJSON operation although looking at what we have here I'm pretty sure you could just re-use the JSON operation.

All of the path logic you have that is custom to polly you could just do in the `defp request` function you have inside `ExAws.Polly`

Gringill commented:
Hey Ben thanks for looking over this, I appreciate it. Your suggested changes sound good 👍🏼 .

I actually originally went down the path of using the JSON operation but felt I was unable to because of the `synthesize_speech` method, the body it returns is binary and as such the json decode that is attempted in the JSON operation's `parse` function fails. Do you have any suggestions regarding that?

Additionally, the `synthesize_speech` method accepts a param `LexiconNames` that is an array of strings which, when passed, causes us to bump into the issue of `encode_query/1` not being able to encode values that are lists.

Gringill commented:
I addressed your concerns about the `ExAws.Polly` module.

Take a look at the JSON operation now. Due to the issue of `synthesize_speech` being the only method that should not be json decoded I took a stab at allowing the parser used to be optionally overridden by the operations parser field. Let me know if you would prefer a different approach.

I now realize that correctly using the JSON operation (as opposed to the since deleted polly operation) causes an actual json body to be sent rather than just encoding the params into the url, which causes the aforementioned `encode_query/1` issue to be side-stepped, hooray.

Gringill commented:
Hey @benwilson512 when you get a chance I would love to make any further changes to this PR that you see fit.

isaiahdw commented:
Any thoughts on getting this merged? I've been using the `Gringill:feature/polly` fork, but it would be great to have this merged and include in an official release.

benwilson512 commented:
This looks good, thanks!

benwilson512 commented on lib/ex_aws/polly.ex:
```diff
@@ -0,0 +1,115 @@
+  end
+
+  defp request(action, data, opts \\ %{}) do
+    operation =
```
This variable is unused, why is it here?

Gringill replied:
Holdover from when I thought I was going to need a `"x-amz-target"` header, I was mistaken.

benwilson512 commented on lib/ex_aws/polly.ex:
```diff
@@ -0,0 +1,115 @@
+
+  import ExAws.Utils, only: [camelize_keys: 1, identity: 2]
+
+  @namespace "Polly_20160610"
```
This attribute is unused

Gringill replied:
Indeed, switching to `@version "v1"` and actually using it.

benwilson512 commented on lib/ex_aws/polly.ex:
```diff
@@ -0,0 +1,115 @@
+    parser =
+      fn
+        {:error, _} = error, _ -> error
+        {:ok, %{body: body, headers: headers}}, config ->
```
This variable is unused.

benwilson512 commented on lib/ex_aws/polly.ex:
```diff
@@ -0,0 +1,115 @@
+    request(:synthesize_speech, opts, request_opts)
+  end
+
+  defp request(action, data, opts \\ %{}) do
```
Defaults are never used.

benwilson512 commented:
Additionally, the test suite does not pass.

Gringill commented:
Thanks for looking at this Ben. I addressed all of your concerns and made some improvements on my feature/polly branch, should I make another pull request?

benwilson512 commented:
yes please!
